src/ApartmentMigration.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace BuildEmpire\Apartment;
+
+use Illuminate\Database\Migrations\Migration;
+
+class ApartmentMigration extends Migration
+{
+    protected $schemas = [];
+    protected $fileName = false;
+    protected $originalTable;
+
+    /**
+     * ApartmentMigration constructor.
+     */
+    public function __construct()
+    {
+        $schema = new Schema();
+        $this->schemas = $schema->getAllSchemas();
+        $this->fileName = $this->getMigrationFileName();
+        $this->originalTable = $this->table;
+    }
+
+    /**
+     * Run/Up all apartment migrations.
+     *
+     * @param bool $ranViaArtisan
+     */
+    public function up($ranViaArtisan = true) {
+        foreach($this->schemas as $schema) {
+
+            $schemaTablePrefix = $schema->name . '.';
+
+            if (!$ranViaArtisan && !$this->hasPublicMigrationRan()) {
+                 continue;
+            }
+
+            if ($this->hasSchemaMigrationRan($schemaTablePrefix)) {
+                continue;
+            }
+
+            $this->setSchemaTable($schemaTablePrefix);
+            $this->updateSchemaMigrateUp($schemaTablePrefix);
+            $this->apartmentUp();
+
+        }
+    }
+
+    /**
+     * Run/Reset all apartment migrations.
+     */
+    public function down() {
+        foreach($this->schemas as $schema) {
+
+            $schemaTablePrefix = $schema->name . '.';
+
+            if (!$this->hasSchemaMigrationRan($schemaTablePrefix)) {
+                continue;
+            }
+
+            $this->setSchemaTable($schemaTablePrefix);
+            $this->updateSchemaMigrateDown($schemaTablePrefix);
+            $this->apartmentDown();
+        }
+    }
+
+    /**
+     * Does the public migration have this migration file.
+     *
+     * @return bool
+     */
+    protected function hasPublicMigrationRan() {
```

Comment:
The migration concept is kinda hard to wrap your head around. One in the schema and one in the public schema. I placed them in two methods to make it clear to the developer.